src/schema.graphql

```diff
@@ -47,6 +49,11 @@ input CreatePostInput {
   body: String
 }
 
+input UpdateCharacterInput {
```

Comment:
Is it necessary to define a new schema for each function payload? For example in the `login` method above we just need to pass function args straight to the function, rather than values in an object payload as the "data" parameter.

Reply:
It's not necessary, but it's a good practice if you ever end up making mutations that take in the same parameters. In this case however, it's unlikely that we'd ever have any other mutations that take the exact same parameters as the UpdateCharacter mutation, so you're right in thinking it would be better to not use an input type here. I will change that

Reply:
Yes I think its better to create classes/schemas when there is actually duplicated code.

Otherwise, if we were to do this for every function signature we are essentially just creating a high coupling problem for ourselves where the function signature is effectively duplicated in two places...

Reply:
Actually, I've looked into this more, and apparently it's a better GraphQL practice to be using input types for mutations, since it becomes a huge pain for writing queries on the client side if you don't do this.

Look here for an explanation: https://www.apollographql.com/blog/designing-graphql-mutations-e09de826ed97/

Also, for an example of this at play in our own codebase, look at `opertations.js` within `tests` > `utils`. These are examples of GraphQL queries on the client side, and you can see right away that the login mutation is way messier, as it doesn't use input types.

Reply:
Speaking of which, I think we should convert the login mutation to use an input type as well to align with these best practices

Reply:
Hm... I would actually strongly disagree with the article linked only if the function signature only appeared once client side, since it is better to be explicit with what the function parameters are at the location of the code itself...

However, because it appears twice and looks shteuuupid I'm going to say that I think both ways are equally...bad. 

So I suppose we might as well...just follow the industry standard here.

Reply:
What do you mean by function signature? I'm having a hard time trying to understand what you're saying because I'm unfamiliar with that term and it doesn't seem to be the proper term to use in the context of GraphQL

Reply:
Oh I think I understand, you're referring to the parameters being used within `MyMutation` as well as `updatePost` in the article? Then yes, I agree that if it only appeared once it wouldn't be so bad. 

However, I have to disagree with you saying that they're both equally bad, using data inputs are for sure superior. Using data inputs reduces the length of graphql queries and mutations written on the client side, and it in fact reduces coupling rather than increases it like you thought.

For example, look at the login mutation again in the file previously mentioned. What happens if I have to make a change where the mutation only accepts emails for logging in, instead of either an email or username? First, I'd have to change the parameters in `schema.graphql`, and on top of that, I'd have to change the parameters again in the client side (in `operations.js`).

However, if I were to use an input type instead, i.e `LoginInput`, I would only have to make this change in the `schema.graphql` file, and the client side code would already be using this `LoginInput`, so I wouldn't have to make any changes there.

Reply:
Hm.

Reply:
Yes.